File: training/2025-01/CodeForces998Div3/G.py
def solve():
    n = int(input())
    val = []
    first_vals = list(map(int, input().split()))
    second_vals = list(map(int, input().split()))
    for i in range(n):
        val.append((first_vals[i], second_vals[i]))
    val.sort(key=lambda x: min(x[0], x[1]))

    dp = [[False]*5 for _ in range(n)]
    dp[0][1] = True
    dp[0][4] = True
    for i in range(1, n):
        if val[i-1][0] < val[i][0] and val[i-1][1] < val[i][1]:
            dp[i][1] |= dp[i-1][1]
            dp[i][2] |= dp[i-1][4]
            dp[i][3] |= dp[i-1][3]
            dp[i][4] |= dp[i-1][2]
        if val[i-1][0] < val[i][1] and val[i-1][1] < val[i][0]:
            dp[i][1] |= dp[i-1][2]
            dp[i][2] |= dp[i-1][3]
            dp[i][3] |= dp[i-1][4]
            dp[i][4] |= dp[i-1][1]

    print("YES" if (dp[n-1][1] or dp[n-1][2]) else "NO")


if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        solve()

File: training/2025-01/CodeForces998Div3/G.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define pll pair<ll, ll>
const ll INF = 1ll << 62;
void solve() {
  ll n;
  cin >> n;
  vector<pll> val(n);
  vector<array<bool, 5>> dp(n);
  for (auto i = 0; i < n; ++i)
    cin >> val[i].first;
  for (auto j = 0; j < n; ++j)
    cin >> val[j].second;
  sort(val.begin(), val.end(), [](pll lhs, pll rhs) {
    return min(lhs.first, lhs.second) < min(rhs.first, rhs.second);
  });
  dp[0][1] = dp[0][4] = true;
  for (auto i = 1; i < n; ++i) {
    if (val[i - 1].first < val[i].first && val[i - 1].second < val[i].second) {
      dp[i][1] |= dp[i - 1][1];
      dp[i][2] |= dp[i - 1][4];
      dp[i][3] |= dp[i - 1][3];
      dp[i][4] |= dp[i - 1][2];
    }
    if (val[i - 1].first < val[i].second && val[i - 1].second < val[i].first) {
      dp[i][1] |= dp[i - 1][2];
      dp[i][2] |= dp[i - 1][3];
      dp[i][3] |= dp[i - 1][4];
      dp[i][4] |= dp[i - 1][1];
    }
  }
  cout << ((dp[n - 1][1] || dp[n - 1][2]) ? "YES" : "NO") << endl;
}
int main() {
  ll test_cases;
  ios::sync_with_stdio(false);
  cin >> test_cases;
  while (test_cases--) {
    solve();
  }
  return 0;
}
